src/wmtk/simplex/link_condition.cpp

```diff
@@ -0,0 +1,90 @@
+#include "link_condition.hpp"
+#include "link.hpp"
+#include "open_star.hpp"
+
+namespace wmtk::simplex {
+bool link_condition_closed_trimesh(const TriMesh& mesh, const Tuple& edge)
+{
+    SimplexCollection lnk_a = link(mesh, Simplex(PrimitiveType::Vertex, edge), true); // link(a)
+    SimplexCollection lnk_b = link(
+        mesh,
+        Simplex(PrimitiveType::Vertex, mesh.switch_tuple(edge, PrimitiveType::Vertex)),
+        true); // link(b)
+    SimplexCollection lnk_ab = link(mesh, Simplex(PrimitiveType::Edge, edge), true); // link(ab)
+
+    SimplexCollection lnk_a_lnk_b_intersection = SimplexCollection::get_intersection(lnk_a, lnk_b);
+
+    return SimplexCollection::are_simplex_collections_equal(lnk_a_lnk_b_intersection, lnk_ab);
+}
+
+bool link_condition(const EdgeMesh& mesh, const Tuple& edge)
+{
+    const Tuple edge_switch_v = mesh.switch_tuple(edge, PrimitiveType::Vertex);
+    if (mesh.is_boundary_vertex(edge) && mesh.is_boundary_vertex(edge_switch_v)) {
```

Comment:
I feel like this should not be part of the link condition. I already mentioned it yesterday when we talked. Let's discuss that tomorrow during our group meeting. Just keep it here for now but I feel like it needs to be moved to some other place.

Reply:
We will keep that as it is for now and probably later on create a second more simple "link condition" (or whatever it will be called).